Bottom: 04d7abec467009420b0c26b4cb045d05e38182cb
Top:    de4f4ec6d1cd9cb2cde075f2ea3f5a71d4445e98
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-06 21:11:38 +0000

find.c: Refactor some common sequences into a helper function


---

diff --git a/src/find.c b/src/find.c
index aee3992..33beba0 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3324,6 +3324,15 @@ struct drc_info
   int flag;
 };
 
+static void
+start_do_it_and_dump (int type, void *ptr1, void *ptr2, void *ptr3,
+                      int flag, bool AndDraw)
+{
+  ListStart (type, ptr1, ptr2, ptr3, flag);
+  DoIt (flag, true, AndDraw);
+  DumpList ();
+}
+
 /*-----------------------------------------------------------------------------
  * Check for DRC violations on a single net starting from the pad or pin
  * sees if the connectivity changes when everything is bloated, or shrunk
@@ -3341,10 +3350,8 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
   if (PCB->Shrink != 0)
     {
       Bloat = -PCB->Shrink;
-      ListStart (What, ptr1, ptr2, ptr3, DRCFLAG | SELECTEDFLAG);
-      DoIt (DRCFLAG | SELECTEDFLAG, true, false);
+      start_do_it_and_dump (What, ptr1, ptr2, ptr3, DRCFLAG | SELECTEDFLAG, false);
       /* ok now the shrunk net has the SELECTEDFLAG set */
-      DumpList ();
       ListStart (What, ptr1, ptr2, ptr3, FOUNDFLAG);
       Bloat = 0;
       drc = true;               /* abort the search if we find anything not already found */
@@ -3356,14 +3363,10 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
           User = true;
           drc = false;
           Bloat = -PCB->Shrink;
-          ListStart (What, ptr1, ptr2, ptr3, SELECTEDFLAG);
-          DoIt (SELECTEDFLAG, true, true);
-          DumpList ();
-          ListStart (What, ptr1, ptr2, ptr3, FOUNDFLAG);
+          start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, true);
           Bloat = 0;
           drc = true;
-          DoIt (FOUNDFLAG, true, true);
-          DumpList ();
+          start_do_it_and_dump (What, ptr1, ptr2, ptr3, FOUNDFLAG, true);
           User = false;
           drc = false;
           drcerr_count++;
@@ -3396,9 +3399,7 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
   drc = false;
   ClearFlagOnAllObjects (false, FOUNDFLAG | SELECTEDFLAG);
   Bloat = 0;
-  ListStart (What, ptr1, ptr2, ptr3, SELECTEDFLAG);
-  DoIt (SELECTEDFLAG, true, false);
-  DumpList ();
+  start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, false);
   flag = FOUNDFLAG;
   ListStart (What, ptr1, ptr2, ptr3, flag);
   Bloat = PCB->Bloat;
@@ -3411,14 +3412,10 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
       User = true;
       drc = false;
       Bloat = 0;
-      ListStart (What, ptr1, ptr2, ptr3, SELECTEDFLAG);
-      DoIt (SELECTEDFLAG, true, true);
-      DumpList ();
-      ListStart (What, ptr1, ptr2, ptr3, FOUNDFLAG);
+      start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, true);
       Bloat = PCB->Bloat;
       drc = true;
-      DoIt (FOUNDFLAG, true, true);
-      DumpList ();
+      start_do_it_and_dump (What, ptr1, ptr2, ptr3, FOUNDFLAG, true);
       drcerr_count++;
       LocateError (&x, &y);
       BuildObjectList (&object_count, &object_id_list, &object_type_list);
@@ -3446,9 +3443,7 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
       /* highlight the rest of the encroaching net so it's not reported again */
       flag = FOUNDFLAG | SELECTEDFLAG;
       Bloat = 0;
-      ListStart (thing_type, thing_ptr1, thing_ptr2, thing_ptr3, flag);
-      DoIt (flag, true, true);
-      DumpList ();
+      start_do_it_and_dump (thing_type, thing_ptr1, thing_ptr2, thing_ptr3, flag, true);
       drc = true;
       Bloat = PCB->Bloat;
       ListStart (What, ptr1, ptr2, ptr3, flag);
